Update FilterEffect::apply() to take in a span instead of a `const Vector&`

cdumez opened the pull request:
#### 9c5696c73fd5c8e2563e78fe679a095effaf411d
<pre>
Update FilterEffect::apply() to take in a span instead of a `const Vector&amp;`
<a href="https://bugs.webkit.org/show_bug.cgi?id=292906">https://bugs.webkit.org/show_bug.cgi?id=292906</a>

Reviewed by Darin Adler.

Update FilterEffect::apply() to take in a span instead of a `const Vector&amp;`. This
avoids having to construct a Vector at one of the call sites.

Source/WebCore/platform/graphics/filters/FilterEffect.cpp:
(WebCore::FilterEffect::takeImageInputs):
Optimize function by leveraging the `Vector(size_t, Generator)` constructor.
This is a bit more efficient as it avoids the capacity checks in append().
Also avoid repeatedly calling takeLast() on the stack and instead just move
the value out of the stack and call `shrink()` once at the end.

Canonical link: <a href="https://commits.webkit.org/294878@main">https://commits.webkit.org/294878@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ee295676ebc12955c97791ae1c8f399a5bb04bf7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103326 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23002 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13323 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/108499 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/53969 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/105365 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/23342 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31553 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78523 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35459 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106332 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18093 "layout-tests (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93215 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58856 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17930 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/11256 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53325 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/87731 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11318 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110874 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30464 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/22493 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87520 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30829 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89412 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87157 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22198 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32017 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9739 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/24798 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/30393 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/35712 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/30199 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/33526 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31761 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/graphics/filters/FilterEffect.cpp:
```diff
@@ -61,28 +61,24 @@ FilterImageVector FilterEffect::takeImageInputs(FilterImageVector& stack) const
+    size_t index = stack.size() - 1;
+    FilterImageVector result(inputsSize, [&](size_t) -> Ref<FilterImage>&& {
+        return WTFMove(stack[index--]);
+    });
```
```suggestion
    size_t index = stack.size();
    FilterImageVector result(inputsSize, [&](size_t) -> Ref<FilterImage>&& {
        return WTFMove(stack[--index]);
    });
```

cdumez committed:
Update FilterEffect::apply() to take in a span instead of a `const Vector&`
https://bugs.webkit.org/show_bug.cgi?id=292906

Reviewed by Darin Adler.

Update FilterEffect::apply() to take in a span instead of a `const Vector&`. This
avoids having to construct a Vector at one of the call sites.

Source/WebCore/platform/graphics/filters/FilterEffect.cpp:
(WebCore::FilterEffect::takeImageInputs):
Optimize function by leveraging the `Vector(size_t, Generator)` constructor.
This is a bit more efficient as it avoids the capacity checks in append().
Also avoid repeatedly calling takeLast() on the stack and instead just move
the value out of the stack and call `shrink()` once at the end.

Canonical link: https://commits.webkit.org/294878@main
